src/include/network/lan/address.rb

```diff
@@ -1005,12 +960,12 @@ def AddressDialog
         ]
       )
 
-      wd["IPOIB_MODE"] = ipoib_mode_widget if LanItems.GetCurrentType == "ib"
+      wd["IPOIB_MODE"] = ipoib_mode_widget if builder.type == "ib"
```

Comment:
this start smelling to me. This type should be enum or own class. But something for future. As plain text comparison can be silently failing for long type. ( I made types often ).

Reply:
definitely. But we don't have type hierarchy yet. Just one simple Interface class.